### PR TITLE
Upd. TRP. Improved adaptive.

### PR DESCRIPTION
**Pull request by svfcode**
task https://doboard.com/1/task/11581

### CODE REVIEW

**Review by AntonV1211 — APPROVED** [review]

File: js/src/apbct-public--7--trp.js [hunk 1]
```diff
@@ -70,4 +70,73 @@ document.addEventListener('DOMContentLoaded', function() {
...
         element.append(trpDescription);
     });
+
+    const badges = document.querySelectorAll('.apbct-real-user-badge');
+
+    badges.forEach((badge) => {
+        let hideTimeout = undefined;
+
+        badge.addEventListener('click', function() {
+            const popup = this.nextElementSibling;
```
Comment (alexandergull) [inline js/src/apbct-public--7--trp.js:80]: https://developer.mozilla.org/en-US/docs/Web/API/Element/nextElementSibling

popup can be null, handle this case

File: js/src/apbct-public--7--trp.js [hunk 2]
```diff
@@ -70,4 +70,73 @@ document.addEventListener('DOMContentLoaded', function() {
...
+                    popup.classList.remove('visible');
+                }
+            }, 1000);
+        });
+
+        // For mobile devices
+        badge.addEventListener('touchend', function() {
+            hideTimeout = setTimeout(() => {
+                const popup = this.nextElementSibling;
+                const selection = window.getSelection();
```
Comment (alexandergull) [inline js/src/apbct-public--7--trp.js:120]: https://developer.mozilla.org/en-US/docs/Web/API/Window/getSelection

selection can return null

**Review by alexandergull — CHANGES_REQUESTED** [review]

File: js/src/apbct-public--7--trp.js (hunk 2 above)
Comment (svfcode) [reply js/src/apbct-public--7--trp.js:120]: nice catch, thanks, fixed https://github.com/CleanTalk/wordpress-antispam/pull/501/commits/f26fe6e5d8f22fd7b99a34e3f5b700ed5224f76a

File: js/src/apbct-public--7--trp.js (hunk 1 above)
Comment (svfcode) [reply js/src/apbct-public--7--trp.js:80]: nice catch, thanks, fixed https://github.com/CleanTalk/wordpress-antispam/pull/501/commits/f26fe6e5d8f22fd7b99a34e3f5b700ed5224f76a

**Pull request merged**